packages/framework/block-std/src/event/dispatcher.ts

```diff
@@ -227,6 +240,43 @@ export class UIEventDispatcher {
     return output;
   }
 
+  private _isActiveElementOutsideHost(): boolean {
+    const activeElement = this._getDeepActiveElement();
+    return (
+      activeElement !== null &&
+      this._isEditableElementActive(activeElement) &&
+      !this.host.contains(activeElement)
+    );
+  }
+
+  private _isEditableElementActive(element: Element | null): boolean {
+    if (!element) return false;
+    return (
+      element instanceof HTMLInputElement ||
+      element instanceof HTMLTextAreaElement ||
+      (element as HTMLElement).isContentEditable
+    );
+  }
+
+  private _setActive(active: boolean) {
+    if (this.host.doc.readonly) return;
```

Comment:
This line will make edgeless canvas can not be moved by touch in mobile readonly mode. So what are the considerations behind this line of change?

Reply:
The consideration here is that in addition to the main editable editor, a web page may also have some read-only preview editors embedded in the sidebar or main editor, and we don't want these read-only editors to be in the active state. However, if this will cause the mobile terminal to be unable to move the viewport when it is read-only, then some modifications may need to be made.

Reply:
Since some checks have been done in certain scenarios：

```
  private _isEditableElementActive(element: Element | null): boolean {
    if (!element) return false;
    return (
      element instanceof HTMLInputElement ||
      element instanceof HTMLTextAreaElement ||
      (element as HTMLElement).isContentEditable
    );
  }
```

And different scenarios have been tested, this line of code can be removed.